Source/SwiftLintFramework/Rules/Lint/HashFunctionRule.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import SourceKittenFramework
+
+public struct HashFunctionRule: ASTRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "hash_function",
+        name: "Hash Function",
+        description: "The new hash function should be preferred in general. " +
+        "Consider using `func hash(into hasher: inout Hasher)` instead.",
+        kind: .lint,
+        minSwiftVersion: .fourDotTwo,
+        nonTriggeringExamples: [
+            """
+            struct Foo: Hashable {
```

Comment:
It'd be great to have some more examples just to be sure (e.g. a global var called `hashValue`, `hashValue` with another type, etc)

Also, a small nitpick: maybe use a `class` as `struct`s now automagically get `Hashable` implementation for free?

Reply:
I think we should keep the struct as some people override it and will implement a custom hash function.